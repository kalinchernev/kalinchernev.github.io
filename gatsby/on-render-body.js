'use strict';

const React = require('react');
const siteConfig = require('../config.js');

// eslint-disable-next-line import/no-webpack-loader-syntax, import/no-unresolved
const katexStylesheet = require('!css-loader!../static/css/katex/katex.min.css');

const onRenderBody = ({ setHeadComponents }) => {
  const { useKatex } = siteConfig;

  if (useKatex) {
    setHeadComponents([
      React.createElement('style', {
        key: 'katex-inline-stylesheet',
        dangerouslySetInnerHTML: { __html: katexStylesheet.toString() },
      }),
    ]);
  }
};

module.exports = onRenderBody;
