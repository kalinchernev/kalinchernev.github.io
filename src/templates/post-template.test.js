// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import PostTemplate from './post-template';
import siteMetadata from '../../jest/__fixtures__/site-metadata';
import markdownRemark from '../../jest/__fixtures__/markdown-remark';
import type { RenderCallback } from '../types';

describe('PostTemplate', () => {
  const props = {
    data: {
      ...markdownRemark,
    },
  };

  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => render(siteMetadata),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  it('renders correctly', () => {
    const tree = renderer.create(<PostTemplate {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
