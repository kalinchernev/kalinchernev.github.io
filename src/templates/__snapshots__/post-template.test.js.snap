// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostTemplate renders correctly 1`] = `
<div
  className="layout"
>
  <div
    className="post"
  >
    <a
      className="post__home-button"
      href="/"
    >
      All Articles
    </a>
    <div
      className="post__content"
    >
      <div
        className="content"
      >
        <h1
          className="content__title"
        >
          test
        </h1>
        <div
          className="content__body"
          dangerouslySetInnerHTML={
            Object {
              "__html": "<p>test</p>",
            }
          }
        />
      </div>
    </div>
    <div
      className="post__footer"
    >
      <div
        className="meta"
      >
        <p
          className="meta__date"
        >
          Published 
          1 Sep 2016
        </p>
      </div>
      <div
        className="tags"
      >
        <ul
          className="tags__list"
        >
          <li
            className="tags__list-item"
          >
            <a
              className="tags__list-item-link"
              href="/test_0"
            >
              test_0
            </a>
          </li>
          <li
            className="tags__list-item"
          >
            <a
              className="tags__list-item-link"
              href="/test_1"
            >
              test_1
            </a>
          </li>
        </ul>
      </div>
      <div
        className="author"
      >
        <p
          className="author__bio"
        >
          Test bio
          <a
            className="author__bio-twitter"
            href="https://www.twitter.com/#"
            rel="noopener noreferrer"
            target="_blank"
          >
            <strong>
              Test name
            </strong>
             on Twitter
          </a>
        </p>
      </div>
    </div>
    <div
      className="post__comments"
    />
  </div>
</div>
`;
